tests/ztest/include/ztest_assert.h

```diff
@@ -29,17 +30,27 @@ static inline void _zassert_(int cond, const char *file, int line)
 	}
 }
 
-#define _zassert(cond, msg, default_msg, file, line, func) \
+#define _zassert(cond, default_msg, file, line, func, msg, ...) \
 	_zassert_(cond, file, line)
 
 #else /* CONFIG_ZTEST_ASSERT_VERBOSE != 0 */
 
-static inline void _zassert(int cond, const char *msg, const char *default_msg,
-			   const char *file, int line, const char *func)
+static inline void _zassert(int cond, const char *default_msg,
```

Comment:
this and any other changes to C zassert APIs need the __printf_like() annotation added for compiler checking

Reply:
Done